File: SDKs/googleapis.com/bigtableadmin/v2/python/src/sdk/models/shared/singleclusterrouting.py
from dataclasses import dataclass, field
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils


@dataclass_json
@dataclass
class SingleClusterRouting:
    r"""SingleClusterRouting
    Unconditionally routes all read/write requests to a specific cluster. This option preserves read-your-writes consistency but does not improve availability.
    """
    
    allow_transactional_writes: Optional[bool] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('allowTransactionalWrites') }})
    cluster_id: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('clusterId') }})
    

File: SDKs/googleapis.com/bigtableadmin/v2/typescript/src/sdk/models/shared/singleclusterrouting.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



// SingleClusterRouting
/** 
 * Unconditionally routes all read/write requests to a specific cluster. This option preserves read-your-writes consistency but does not improve availability.
**/
export class SingleClusterRouting extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=allowTransactionalWrites" })
  allowTransactionalWrites?: boolean;

  @SpeakeasyMetadata({ data: "json, name=clusterId" })
  clusterId?: string;
}
